// Rstest Snapshot v1

exports[`logger > should create new logger with info level correctly 1`] = `
[
  [
    "ðŸ˜Š Rslog v1.0.0
",
  ],
  [
    "this is a log message",
  ],
  [
    "error   this is a error message",
  ],
  [
    "info    this is an info message",
  ],
  [
    "warn    this is a warn message",
  ],
  [
    "ready   this is a ready message",
  ],
  [
    "success this is a success message",
  ],
]
`;

exports[`logger > should create new logger with warn level correctly 1`] = `
[
  [
    "error   this is a error message",
  ],
  [
    "warn    this is a warn message",
  ],
]
`;

exports[`logger > should log as expected 1`] = `
[
  [
    "ðŸ˜Š Rslog v1.0.0
",
  ],
  [
    "this is a log message",
  ],
  [
    "error   this is a error message",
  ],
  [
    "info    this is an info message",
  ],
  [
    "warn    this is a warn message",
  ],
  [
    "ready   this is a ready message",
  ],
  [
    "success this is a success message",
  ],
]
`;

exports[`logger > should log error with cause correctly 1`] = `
error   this is an error message with cause
    at <ROOT>/tests/logger.test.ts:87:7
    at Object.fn (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:578:25)
    at runTestsCase (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:691:40)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:808:51)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:786:25)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at TestRunner.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:842:17)
    at Object.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:1304:41)
    at runInPool (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/worker.js:5957:25)
  [cause]: this is a cause error
    at getErrorCause (<ROOT>/tests/logger.test.ts:16:10)
    at <ROOT>/tests/logger.test.ts:88:16
    at Object.fn (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:578:25)
    at runTestsCase (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:691:40)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:808:51)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:786:25)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at TestRunner.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:842:17)
    at Object.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:1304:41)
`;

exports[`logger > should log error with stack correctly 1`] = `
error   this is an error message
    at <ROOT>/tests/logger.test.ts:78:18
    at Object.fn (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:578:25)
    at runTestsCase (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:691:40)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:808:51)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at runTest (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:786:25)
    at runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:739:25)
    at TestRunner.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:842:17)
    at Object.runTests (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/867.js:1304:41)
    at runInPool (file://<ROOT>/node_modules/.pnpm/@rstest+core@0.1.0/node_modules/@rstest/core/dist/worker.js:5957:25)
`;
